e2e/test/iothub/service/RegistryManagerExportDevicesTests.cs

```diff
@@ -101,123 +103,222 @@ await registryManager
                         })
                     .ConfigureAwait(false);
 
+                Configuration configuration = await registryManager
+                    .AddConfigurationAsync(
+                        new Configuration(configurationId)
+                        {
+                            Priority = 2,
+                            Labels = { { "labelName", "labelValue" } },
+                            TargetCondition = "*",
+                            Content =
+                            {
+                                DeviceContent = { { "properties.desired.x", 4L } },
+                            },
+                            Metrics =
+                            {
+                                Queries = { { "successfullyConfigured", "select deviceId from devices where properties.reported.x = 4" } }
+                            },
+                        })
+                    .ConfigureAwait(false);
+
                 // act
 
-                JobProperties exportJobResponse = null;
-                int tryCount = 0;
-                while (true)
+                JobProperties exportJobResponse = await RunExportJobAsync(
+                        storageAuthenticationType,
+                        isUserAssignedMsi,
+                        devicesFileName,
+                        configsFileName,
+                        registryManager,
+                        containerUri)
+                    .ConfigureAwait(false);
+
+                exportJobResponse = await WaitForJobToCompleteAsync(registryManager, exportJobResponse).ConfigureAwait(false);
+
+                // assert
+                await ValidateDevicesAsync(
+                        devicesFileName,
+                        storageContainer,
+                        edge1,
+                        edge2,
+                        device)
+                .ConfigureAwait(false);
+                await ValidateConfigurationsAsync(
+                        configsFileName,
+                        storageContainer,
+                        configuration)
+                    .ConfigureAwait(false);
+            }
+            finally
+            {
+                await CleanUpDevicesAsync(edgeId1, edgeId2, deviceId, configurationId, registryManager).ConfigureAwait(false);
+            }
+        }
+
+        private async Task<JobProperties> RunExportJobAsync(StorageAuthenticationType storageAuthenticationType, bool isUserAssignedMsi, string devicesFileName, string configsFileName, RegistryManager registryManager, Uri containerUri)
```

Comment:
The method above was getting pretty long, so I broke out running the job to a separate method.